src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -520,7 +526,10 @@ service::service(
   , _auth{
       config::always_true(),
       config::shard_local_cfg().superusers.bind(),
-      controller.get()} {}
+      controller.get()} {
+    _inflight_config_binding.watch(
+      [this]() { _inflight_sem.set_capacity(_inflight_config_binding()); });
```

Comment:
`_inflight_sem` could be destructed at this point (I don't see another mechanism that unsubscribes the watch, or otherwise sequences the destruction order.

Reply:
Isn't destruction sequence order dictated by the class?  `_inflight_config_binding` is constructed after `_inflight_sem` and therefor should be destructed before the semaphore?

Reply:
BTW I find relying on destructor order is incredibly sneaky and usually deserves a comment somewhere

Reply:
> Isn't destruction sequence order dictated by the class? `_inflight_config_binding` is constructed after `_inflight_sem` and therefor should be destructed before the semaphore?

You're correct.